src/main/java/apoc/util/Util.java

```diff
@@ -700,4 +705,44 @@ public static boolean classExists(String className) {
         Object value = ApocConfiguration.get(loadType).get(key);
         return Optional.ofNullable(value).map(Object::toString);
     }
+
+    public static String formatDate(TemporalAccessor value, String format){
+        return getFormat(format).format(value);
+    }
+
+    public static TemporalAccessor parseDate(String value, Class<? extends TemporalAccessor> date, String...formats) {
+        String methodName = "parse";
+        try {
+            Method method = date.getDeclaredMethod(methodName, CharSequence.class, DateTimeFormatter.class);
+            Method methodSimple = date.getDeclaredMethod(methodName, CharSequence.class);
+
+            if (formats != null && formats.length > 0) {
+                for (String form : formats) {
+                    try {
+                        try {
+                            return (TemporalAccessor) method.invoke(null, value, getFormat(form));
+                        } catch (InvocationTargetException e) {
+                            return (TemporalAccessor) methodSimple.invoke(null, value);
+                        }
+                    } catch (Exception e) {
+                        continue;
+                    }
+                }
+            } else {
+                return (TemporalAccessor) methodSimple.invoke(null, value);
+            }
+
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+        throw new RuntimeException("Can't format the date with the pattern");
+    }
+
+    public static Duration parseDuration(String value) {
+        return Duration.parse(value);
+    }
+
+    private static DateTimeFormatter getFormat(String format) {
+        return ISO_DATE_FORMAT.computeIfAbsent(format.toLowerCase(), k -> DateTimeFormatter.ofPattern(format));
```

Comment:
`computeIfAbsent` se non c'è l'elemento lo inserisce questo vuol dire che nel tempo quella variabile potrebbe diventare immensa. Cambiamo così
```suggestion
        if (ISO_DATE_FORMAT.containsKey(format.toLowerCase())) {
                return ISO_DATE_FORMAT.get(format.toLowerCase());
        }
        return DateTimeFormatter.ofPattern(format);
```

Reply:
Usa lo snippet che ti ho mandato per cambiare `ISO_DATE_FORMAT`